### PR TITLE
E-notice fix on Is delete record

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
E-notice fix on Is delete record

Before
----------------------------------------
image

After
----------------------------------------
Erm - I only fixed the top one

Technical Details
----------------------------------------
@stesi561 - you wind up with fixes like this...

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton 

```
Fatal error: Declaration of CRM_Profile_Form_Edit::buildQuickForm() must be compatible with CRM_Profile_Form::buildQuickForm(): void in /home/jenkins/bknix-dfl/build/core-24157-6mrhm/web/sites/all/modules/civicrm/CRM/Profile/Form/Edit.php on line 137
```

File: CRM/Profile/Form.php [hunk 1]
```diff
@@ -689,8 +689,9 @@ public function setDefaultsValues() {
    * Build the form object.
    *
    */
-  public function buildQuickForm() {
+  public function buildQuickForm(): void {
     $this->add('hidden', 'gid', $this->_gid);
+    $this->assign('deleteRecord', $this->isDeleteMode());
```
Comment (demeritcowboy) [inline CRM/Profile/Form.php:694]:
```suggestion
    $this->assign('deleteRecord', $this->_recordExists);
```

Comment (eileenmcnaughton) [reply CRM/Profile/Form.php:694]: I figured that part didn't matter because it would bounce & never display the template if the record doesn't exist

**Comment by eileenmcnaughton** [comment]
I added the `void` to the functions on the child classes  - since it seems at some point we will need them given where php is going

File: CRM/Profile/Form.php (hunk 1 above)
Comment (demeritcowboy) [reply CRM/Profile/Form.php:694]: Ok I see.

**Comment by demeritcowboy** [comment]
> I added the void to the functions on the child classes - since it seems at some point we will need them given where php is going

Yes probably. A search of universe says this will break contactlayouteditor.

Otherwise this runs ok and removes the notice.

**Comment by colemanw** [comment]
I'm not a huge fan of these type hints because of their tendency to break things. In a big universe of extensions extending other classes, I think we'd be safer without them. My 2c.

**Pull request merged**